fairlearn/metrics/_plotter.py

```diff
@@ -0,0 +1,272 @@
+            for ax in axs.flatten():
+                if kind == "scatter":
+                    color = ax.lines[0].get_color()
+                else:
+                    color = "black"
+
+                # extend legend with 95% CI text
+                handles, labels = ax.get_legend_handles_labels()
+                custom_line = [Line2D([0], [0], color=color, label="95% CI")]
+                handles.extend(custom_line)
+                ax.legend(handles=handles)
+        else:
+            if kind == "scatter":
+                color = axs.lines[0].get_color()
+            else:
+                color = "black"
+
+            # extend legend with 95% CI text
+            handles, labels = axs.get_legend_handles_labels()
+            custom_line = [Line2D([0], [0], color=color, label="95% CI")]
+            handles.extend(custom_line)
+            axs.legend(handles=handles)
+
+    else:
+        if kind == "scatter":
+            axs = df[metrics].plot(linestyle='', marker='o', subplots=subplots, **kwargs)
+        else:
+            axs = df[metrics].plot(kind=kind, subplots=subplots, **kwargs)
+
+    return axs
+
+
+def plot_metric_frame(metric_frame: MetricFrame,
+                      kind: str = "scatter",
+                      metrics: Union[List[str], str] = None,
+                      error_bars: Union[List[str], str] = None,
+                      conf_intervals: Union[List[str], str] = None,
+                      subplots: bool = True,
+                      plot_error_labels: bool = False,
+                      text_precision_digits: int = 4,
+                      text_fontsize: int = 8,
+                      text_color: str = "black",
+                      text_ha: str = "center",
+                      **kwargs
+                      ) -> Union[Axes, List[Axes]]:
+    """Visualization for metrics with statistical error bounds.
+
+    Plots a given metric and its given error (as described by the `error_bars` or `conf_intervals`)
+
+    This function takes in a :class:`fairlearn.metrics.MetricFrame` with precomputed metrics
+    and metric errors and a `error_bars` or `conf_intervals` array to interpret the columns
+    of the :class:`fairlearn.metrics.MetricFrame`.
+
+    The items at each index of the given `metrics` array and given `error_bars` or `conf_intervals`
+    array should correspond to a pair of the same metric and metric error, respectively.
+
+    Parameters
+    ----------
+    metric_frame : fairlearn.metrics.MetricFrame
+        The collection of disaggregated metric values, along with the metric errors.
+
+    kind : str, default="scatter"
+        The type of plot to display. i.e. "bar", "line", etc.
+        List of options is detailed in :meth:`pandas.DataFrame.plot`
+
+    metrics : str or list of str
+        The name of the metrics to plot.
+        Should match columns from the given :class:`fairlearn.metrics.MetricFrame`.
+
+    error_bars : str or list of str
```

Comment:
I think that we should rename this to `errors`. I just noticed that currently `error_bars` need to be tuples of size two. I would prefer if `errors` were provided as scalars. I don't think that we need to provide functionality for asymmetric errors, since we allow providing directly confidence intervals. Also, for asymmetric errors, I'd be confused about the sign of the lower error and would think that it should be negative.